troposphere/static/js/components/modals/image_version/ImageSelect.react.js

```diff
@@ -11,7 +11,12 @@ export default React.createClass({
     },
 
     handleChange: function(e){
-      this.props.onChange(e.target.value)
+      let image_id = e.target.value;
+      let username = stores.ProfileStore.get().get('username'),
+          image = stores.ImageStore.fetchWhere({
+          created_by__username: username
+      }).get(image_id);
```

Comment:
I might consider making the statement purpose a bit more clear: 

```javascript
 +      let image_id = e.target.value,
 +          username = stores.ProfileStore.get().get('username'),
 +          image = stores.ImageStore.fetchWhere({
 +            created_by__username: username
 +          }).get(image_id);
```

I don't know how to handle the _double_ `get()` really:
```javascript
let username = stores.ProfileStore.get().get('username')
```

I think this might be an _"accepted"_ pattern in the existing Flux implementation. 

Thoughts @cdosborn ?